### ISSUE TITLE
ModelAdmin defined the custom url segment approach broken

### ISSUE THREAD

**Issue by pack254** [silverstripe/silverstripe-admin#1668]
### Module version(s) affected

1.13.19

### Description

 I have found an issue in `silverstripe/admin` version `1.13.19` when I did the upgrade security patches. this version it breaks the ModelAdmin which is defined using the custom url segment approach e.g.
```
private static $managed_models = [
  'product-category' => [
    'dataClass' => Category::class,
    'title' => 'Product categories',
   ],
]
```
This patch 1.13.19 appears to break potentially caused from line 662 [ModelAdmin.php](https://github.com/silverstripe/silverstripe-admin/compare/1.13.18...1.13.19#diff-d6ec8df426c5d673f7390f98f9bcf37dee0828848957d1903fd10e1b9e99d96dL662) is changed from `modelClass` to `modelTab`.
In a nuthshell, any module that uses this approach will break i.e. [SiteBanner module](https://github.com/silverstripe/silverstripe-sitebanner/blob/master/src/Models/Admin.php#L13)

### How to reproduce

This can be reproduced by using the `silverStripe/admin` version `1.13.19` and `nzta/silverstripe-sitebanner` version `3.0.1` after running `composer install && composer dev-build flush=all` logged in to CMS and then go to Site Banner tab at the left panel, so you will see this error.

![Screen Shot 2024-01-31 at 2 02 12 PM](https://github.com/silverstripe/silverstripe-admin/assets/17997208/e2e97021-7552-4fad-b744-abab76c920bb)

### Possible Solution

revert code back to `$importer = new $importerClass($this->modelClass);`

### Validations

- [x] Check that there isn't already an issue that reports the same bug
- [X] Double check that your reproduction steps work in a fresh installation of [`silverstripe/installer`](https://github.com/silverstripe/silverstripe-installer) (with any code examples you've provided)

## PRs
- https://github.com/silverstripe/silverstripe-admin/pull/1670

**Comment by emteknetnz** [comment]
@pack254 linked PR has been merged, will be auto-tagged shortly

Note I left a comment on https://github.com/silverstripe/silverstripe-admin/pull/1670#pullrequestreview-1857624066 that the site banner possibly isn't working correctly, so you'll want to validate this on your local

Fixing that would be a totally separate issue and it's not a module that the open-source team look after

**Comment by emteknetnz** [comment]
Note this hasn't released just yet because of a CI failure in silverstripe/admin which is being tracked [here](https://github.com/silverstripe/silverstripe-admin/issues/1671)

**Comment by emteknetnz** [comment]
This has now released as [1.13.21](https://github.com/silverstripe/silverstripe-admin/releases/tag/1.13.21)
